container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/kubernetes_kurtosis_backend_enclave_functions.go

```diff
@@ -477,8 +479,46 @@ func (backend *KubernetesKurtosisBackend) getMatchingEnclaveKubernetesResources(
 	}
 
 	// Per-namespace objects
-	result := map[enclave.EnclaveUUID]*enclaveKubernetesResources{}
+	getEnclaveResourcesOperations := map[operation_parallelizer.OperationID]operation_parallelizer.Operation{}
 	for enclaveIdStr, namespacesForEnclaveId := range namespaces {
+		getEnclaveResourcesOperations[operation_parallelizer.OperationID(enclaveIdStr)] = backend.createGetEnclaveResourcesOperation(ctx, namespacesForEnclaveId, enclaveIdStr)
+	}
+	successfulEnclaveResources, failedEnclaveResources := operation_parallelizer.RunOperationsInParallel(getEnclaveResourcesOperations)
+
+	if len(failedEnclaveResources) > 0 {
+		var allOperationsErrors []string
+		var erroredEnclaveUUIDs []enclave.EnclaveUUID
+		for operationID, operationErr := range failedEnclaveResources {
+			enclaveUUID := enclave.EnclaveUUID(operationID)
+			erroredEnclaveUUIDs = append(erroredEnclaveUUIDs, enclaveUUID)
+			operationErrStr := fmt.Sprintf("enclave UUID '%v' - error:%v", enclaveUUID, operationErr.Error())
+			allOperationsErrors = append(allOperationsErrors, operationErrStr)
+		}
+		allOperationsErrorsStr := strings.Join(allOperationsErrors, "\n")
+		return nil, stacktrace.NewError("Running the get enclave Kubernetes resources operations for enclaves with UUIDs '%+v' returned errors. Errors:\n%v", erroredEnclaveUUIDs, allOperationsErrorsStr)
```

Comment:
do we care about any partial successes that have happened so far?

Reply:
Not so far, because the `getMatchingEnclaveKubernetesResources`  doesn't support this, I mean it didn't return successful enclave resources and not successful enclave resources, and if we want to add this, it will necessary to refactor the `GetEnclaves` backend method in order to take advantage of the result, so it will demand a bigger refactor. But I'm happy to create a ticket for it and address it later